pymedphys/_dicom/anonymise/api.py

```diff
@@ -334,32 +340,45 @@ def anonymise_directory(
 
     identifying_keywords: ``list``, optional
         If left as None, the default values for/list of identifying keywords are used
+
+    Returns
+    -------
+    ``list`` of anonymised file paths
     """
     dicom_dirpath = str(dicom_dirpath)
 
     dicom_filepaths = glob(dicom_dirpath + "/**/*.dcm", recursive=True)
     failing_filepaths = []
-    # errors = []
+    successful_filepaths = []
+    anon_filepaths = []
+    errors = []
 
     for dicom_filepath in dicom_filepaths:
         if output_dirpath is not None:
             relative_path = os.path.relpath(dicom_filepath, start=dicom_dirpath)
             output_filepath = os.path.join(output_dirpath, relative_path)
         else:
             output_filepath = None
-
-        anonymise_file(
-            dicom_filepath,
-            output_filepath=output_filepath,
-            delete_original_file=delete_original_files,
-            anonymise_filename=anonymise_filenames,
-            replace_values=replace_values,
-            keywords_to_leave_unchanged=keywords_to_leave_unchanged,
-            delete_private_tags=delete_private_tags,
-            delete_unknown_tags=delete_unknown_tags,
-            replacement_strategy=replacement_strategy,
-            identifying_keywords=identifying_keywords,
-        )
+        try:
+            dicom_anon_filepath = anonymise_file(
+                dicom_filepath,
+                output_filepath=output_filepath,
+                delete_original_file=delete_original_files,
+                anonymise_filename=anonymise_filenames,
+                replace_values=replace_values,
+                keywords_to_leave_unchanged=keywords_to_leave_unchanged,
+                delete_private_tags=delete_private_tags,
+                delete_unknown_tags=delete_unknown_tags,
+                replacement_strategy=replacement_strategy,
+                identifying_keywords=identifying_keywords,
+            )
+            successful_filepaths.append(dicom_filepath)
+            anon_filepaths.append(dicom_anon_filepath)
+        except (AttributeError, LookupError, TypeError, OSError, ValueError) as error:
```

Comment:
I'm okay with using the BaseException / Exception here, and then just overriding the pylint error. Since the error is going to be reraised later, I believe pylint's complaint isn't relevant to this situation.

Reply:
I think it's OK to just have a handful of known errors, and if it's a different error, the error rips up through the stack.
That ends up being no different than what was there previously (which was no try/error at all).

Reply:
>I'm okay with using the BaseException / Exception here

Is that a request/requirement for me to revert and override the pylint error?
I'm fine either way, but if it's just "it would be OK", then I'd rather move forward.
If it's "this must change", I'll go ahead and do that.

Reply:
No, certainly not a must change. I just noticed you went from one to the other, and I wondered if it was to make pylint happy.